misc/futures-bounded/src/stream_set.rs

```diff
@@ -0,0 +1,60 @@
+use futures_util::stream::BoxStream;
+use futures_util::Stream;
+use std::task::{ready, Context, Poll};
+use std::time::Duration;
+
+use crate::{PushError, StreamMap, Timeout};
+
+/// Represents a list of [Stream]s.
+///
+/// Each stream must finish within the specified time and the list never outgrows its capacity.
+pub struct StreamSet<O> {
+    id: u32,
+    inner: StreamMap<u32, O>,
+}
```

Comment:
Thinking about the naming here, I think we should call them `FuturesList` and `StreamList` because they don't enforce any uniqueness that one would expect from a "set".

Reply:
But then they also don't enforce any order one would expect from a "list" ...

Reply:
`FuturesBag` and `StreamBag` perhaps?

Reply:
I have never heard _bag_ used in regards to programming language collections.

_Set_ seems fine to me. One could argue that a `Future` or `Stream` is by definition unique.

As an alternative, I think the generic word _collection_ would do as well.